l10n_ch_base_bank/models/bank.py

```diff
@@ -236,34 +266,53 @@ def _check_postal_num(self):
         return True
 
     @api.multi
-    def _get_acc_name(self):
-        """ Return an account name for a bank account
-        to use with a ccp for ISR.
-        This method makes sure to generate a unique name
-        """
-
+    def _update_acc_name(self):
+        """Check if number generated from ccp, if yes replace it on new """
         part_name = self.partner_id.name
         if not part_name and self.env.context.get('default_partner_id'):
             partner_id = self.env.context.get('default_partner_id')
             part_name = self.env['res.partner'].browse(partner_id)[0].name
+        self.acc_number = self._compute_name_ccp(part_name, self.ccp)
 
-        if part_name:
-            acc_name = _("Bank/CCP {}").format(part_name)
+    @api.multi
+    def _compute_name_ccp(self, partner_name, ccp):
+        """This method makes sure to generate a unique name"""
+        if partner_name and ccp:
+            acc_name = _("{}/CCP {}").format(partner_name, ccp)
+        elif partner_name:
+            acc_name = _("{}/CCP").format(partner_name)
```

Comment:
When emptying acc_number or ccp the acc_number field is again filled.

If we don't have a ccp number it is better to keep it empty the user might not be filling a CCP account.


Thus I would suggest to replace it by:

``` python
        if partner_name and ccp:
            acc_name = _("{}/CCP {}").format(partner_name, ccp)
        elif ccp:
            acc_name = _("CCP {}").format(ccp)
        else:
            return ''
```